src/Arcus.WebApi.Unit/OpenApi/OAuthAuthorizeOperationFilterTests.cs

```diff
@@ -1,18 +1,111 @@
+                using (var responseStream = await response.Content.ReadAsStreamAsync())
+                {
+                    OpenApiDocument swagger = reader.Read(responseStream, out OpenApiDiagnostic diagnostic);
+                    _outputWriter.WriteLine(diagnostic.Errors.Count == 0 ? String.Empty : String.Join(", ", diagnostic.Errors.Select(e => e.Message + ": " + e.Pointer)));
+
+                    Assert.True(
+                        swagger.Paths.TryGetValue("/oauth/authorize", out OpenApiPathItem oauthPath), 
+                        "Cannot find OAuth authorized path in Open API spec file");
+
+                    Assert.True(
+                        oauthPath.Operations.TryGetValue(OperationType.Get, out OpenApiOperation oauthOperation),
+                        "Cannot find OAuth GET operation in Open API spec file");
+
+                    OpenApiResponses oauthResponses = oauthOperation.Responses;
+                    Assert.Contains(oauthResponses, r => r.Key == "401");
+                    Assert.Contains(oauthResponses, r => r.Key == "403");
+                }
+            }
+        }
+
+        [Fact]
+        public async Task OAuthAuthorizeOperationFilter_ShouldNotIncludeSecurityDefinitionResponses_OnNonAuthorizedOperations()
+        {
+            // Arrange
+            using (var client = _testServer.CreateClient())
+                // Act
+            using (HttpResponseMessage response = await client.GetAsync("swagger/v1/swagger.json"))
+            {
+                // Assert
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+                var reader = new OpenApiStreamReader();
+                using (var responseStream = await response.Content.ReadAsStreamAsync())
+                {
+                    OpenApiDocument swagger = reader.Read(responseStream, out OpenApiDiagnostic diagnostic);
+                    _outputWriter.WriteLine(diagnostic.Errors.Count == 0 ? String.Empty : String.Join(", ", diagnostic.Errors.Select(e => e.Message + ": " + e.Pointer)));
+
+                    Assert.True(
+                        swagger.Paths.TryGetValue("/oauth/none", out OpenApiPathItem oauthPath), 
+                        "Cannot find OAuth none authorized path in Open API spec file");
+
+                    Assert.True(
+                        oauthPath.Operations.TryGetValue(OperationType.Get, out OpenApiOperation oauthOperation),
+                        "Cannot find OAuth GET operation in Open API spec file");
+
+                    OpenApiResponses oauthResponses = oauthOperation.Responses;
+                    Assert.DoesNotContain(oauthResponses, r => r.Key == "401");
+                    Assert.DoesNotContain(oauthResponses, r => r.Key == "403");
+                }
+            }
+        }
+
+        // TODO: should we add a test to check if we also include the security definitions when we add one of our authentication filters?
```

Comment:
Why not, good suggestion!

Reply:
Not sure if it can be done, since the authentication operation filter checks 'OPERATIONS' and has no access to global defined filters (as far as I know).

Reply:
Can you elaborate on that please?

Reply:
The `IOperationFilter.Apply` gets the kind of HTTP `Operation` and the operation's 'context'. Which means, the called HTTP method. In this context, we don't have access to any "global" services like in an `ActionFilter` or `AuthenticationFilter`, which means we can't know with the information we get inside the OAuth Authorize Operation if the hosting application uses any kind of authorization filter (shared access, certificate, or maybe another all together ...); 

Therefore, my statement: not sure if we can implement this with the current state.

Reply:
But we have access to `OpenApiDocument` don't we? So can't we get it like that? It should be in there.

Reply:
Yeah; like I said: it can be tested; but maybe not implemented.

Reply:
Oh now I get it. I think we don't have to enforce this and leave it up to the consumer to be honest.

Sorry I was confused. Think this can then remove the TODO and resolve the comment?

Reply:
Yep, will do.